pyanaconda/modules/network/nm_client.py

```diff
@@ -0,0 +1,667 @@
+        if wired_setting:
+            mac = wired_setting.get_mac_address()
+            if mac:
+                iface = get_iface_from_hwaddr(nm_client, mac)
+    return iface
+
+def get_iface_from_hwaddr(nm_client, hwaddr):
+    """Find the name of device specified by mac address."""
+    for device in nm_client.get_devices():
+        if device.get_device_type() in (NM.DeviceType.ETHERNET,
+                                        NM.DeviceType.WIFI):
+            try:
+                address = device.get_permanent_hw_address()
+            except AttributeError as e:
+                log.warning("Device %s: %s", device.get_iface(), e)
+                address = device.get_hw_address()
+        else:
+            address = device.get_hw_address()
+        if address.upper() == hwaddr.upper():
+            return device.get_iface()
+    return None
+
+def get_team_port_config_from_connection(nm_client, uuid):
+    connection = nm_client.get_connection_by_uuid(uuid)
+    if not connection:
+        return None
+    team_port = connection.get_setting_team_port()
+    if not team_port:
+        return None
+    config = team_port.get_config()
+    return config
+
+def get_team_config_form_connection(nm_client, uuid):
+    connection = nm_client.get_connection_by_uuid(uuid)
+    if not connection:
+        return None
+    team = connection.get_setting_team()
+    if not team:
+        return None
+    config = team.get_config()
+    return config
+
+def get_device_name_from_network_data(nm_client, network_data, supported_devices, bootif):
+    """Get the device name from kickstart device specification.
+
+    Generally given by --device option. For vlans also --interfacename
+    and --vlanid comes into play.
+
+    Side effect: for vlan sets network_data.parent value from --device option
+
+    :param network_data: a kickstart device configuartion
+    :type network_data: kickstart NetworkData object
+    :param supported_devices: list of names of supported devices
+    :type supported_devices: list(str)
+    :param bootif: MAC addres of device to be used for
+                    --device=bootif specification
+    :type bootif: str
+    :returns: device name the configuration should be used for
+    :rtype: str
+    """
+    spec = network_data.device
+    device_name = ""
+    msg = ""
+
+    if not spec:
+        msg = "device specification missing"
+
+    # Specification by device name
+    if spec in supported_devices:
+        device_name = spec
+        msg = "existing device found"
+    # Specification by mac address
+    elif ':' in spec:
+        device_name = get_iface_from_hwaddr(nm_client, spec) or ""
+        msg = "existing device found"
+    # Specification by BOOTIF boot option
+    elif spec == 'bootif':
+        device_name = get_iface_from_hwaddr(nm_client, bootif) or ""
+        msg = "existing device for {} found".format(bootif)
+    # First device with carrier (sorted lexicographically)
+    elif spec == 'link':
```

Comment:
I would rather abstract the `spec == ...` parts into a method with meaningful names instead of adding comments everywhere.

Reply:
It seems fine to me in this function, I'd need a concrete suggestion. Also, I'll write (add back IIRC) a unit test for the function first. I did this:  https://github.com/rhinstaller/anaconda/pull/1771/commits/496622c96e9f41f9c6dda09c744eda5b52eba431 though.

Reply:
ahh, you know what when I'm looking on that now with fresh head it seems fine to me. Sorry for giving you extra work.
You can ignore this.